pkg/venafi/cloud/connector.go

```diff
@@ -827,6 +987,96 @@ func (c *Connector) RetrieveSystemVersion() (response string, err error) {
 	panic("operation is not supported yet")
 }
 
+func setProvisioningOptions(options *endpoint.ProvisioningOptions) (*cloudkeystores.CertificateProvisioningOptionsInput, error) {
+	var cloudOptions *cloudkeystores.CertificateProvisioningOptionsInput
+	dataOptions, err := json.Marshal(options)
+	if err != nil {
+		return nil, err
+	}
+
+	graphqlAzureOptions := &cloudkeystores.CertificateProvisioningAzureOptionsInput{}
+	graphqlGCPOptions := &cloudkeystores.CertificateProvisioningGCPOptionsInput{}
+
+	if options != nil {
+		switch (*options).GetType() {
+		case string(cloudkeystores.CloudKeystoreTypeAcm):
+			// nothing
+		case string(cloudkeystores.CloudKeystoreTypeAkv):
+			err = json.Unmarshal(dataOptions, graphqlAzureOptions)
+			if err != nil {
+				return nil, err
+			}
+		case string(cloudkeystores.CloudKeystoreTypeGcm):
+			err = json.Unmarshal(dataOptions, graphqlGCPOptions)
+			if err != nil {
+				return nil, err
+			}
+		default:
+			return nil, fmt.Errorf("unknown cloud keystore type: %s", (*options).GetType())
+		}
+	}
+
+	cloudOptions = &cloudkeystores.CertificateProvisioningOptionsInput{
+		AwsOptions:   nil,
+		AzureOptions: graphqlAzureOptions,
+		GcpOptions:   graphqlGCPOptions,
+	}
+	return cloudOptions, nil
+}
+
+func getCloudMetadataFromWebsocketResponse(respMap interface{}, keystoreType cloudkeystores.CloudKeystoreType, keystoreId string) (*CloudProvisioningMetadata, error) {
+
+	val := CloudKeystoreProvisioningResult{}
+	valJs, err := json.Marshal(respMap)
+	if err != nil {
+		return nil, fmt.Errorf("unable to encode response data! Error: %s", err.Error())
+	}
+	err = json.Unmarshal(valJs, &val)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse response data! Error: %s", err.Error())
+	}
+	if val.Error != nil {
+		return nil, fmt.Errorf("unable to provision certificate! Error: %s", val.Error)
+	}
+
+	if val.CloudProviderCertificateId == "" {
+		return nil, fmt.Errorf("provisioning is not successful, certificate ID from response is empty")
+	}
+
+	cloudMetadata := &CloudProvisioningMetadata{}
+	switch keystoreType {
+	case cloudkeystores.CloudKeystoreTypeAcm:
+		cloudMetadata.awsMetadata.result = val
+	case cloudkeystores.CloudKeystoreTypeAkv:
+		cloudMetadata.azureMetadata.result = val
+	case cloudkeystores.CloudKeystoreTypeGcm:
+		cloudMetadata.gcpMetadata.result = val
+	default:
+		err = fmt.Errorf("Unknown ConnectorType %v found for keystore with ID: %s", keystoreType, keystoreId)
+		return nil, err
+	}
+	return cloudMetadata, err
+}
+
+func validateProvisioningInputs(req endpoint.ProvisioningRequest) error {
+	if req.KeystoreId != nil && (req.KeystoreName != nil || req.ProviderName != nil) {
```

Comment:
I don't think we need to limit users regarding what attributes they pass. Just set a priority. If keystoreID is present ignore the names.

Reply:
I was following this convention as we discussed that's what we do in VCert CLI (prevent user setting wrong info or more than the expected) but I could change to only ignore it the extra info, if that's preffered

Reply:
Resolved by latest commit